hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataReader.java

```diff
@@ -104,19 +105,29 @@ public OmMetadataReader(KeyManager keyManager,
     this.metrics = omMetadataReaderMetrics;
     this.perfMetrics = ozoneManager.getPerfMetrics();
     if (isAclEnabled) {
-      accessAuthorizer = getACLAuthorizerInstance(configuration);
-      if (accessAuthorizer instanceof OzoneNativeAuthorizer) {
-        OzoneNativeAuthorizer authorizer =
-            (OzoneNativeAuthorizer) accessAuthorizer;
-        isNativeAuthorizerEnabled = true;
-        authorizer.setVolumeManager(volumeManager);
-        authorizer.setBucketManager(bucketManager);
-        authorizer.setKeyManager(keyManager);
-        authorizer.setPrefixManager(prefixManager);
-        authorizer.setAdminCheck(ozoneManager::isAdmin);
-        authorizer.setReadOnlyAdminCheck(ozoneManager::isReadOnlyAdmin);
-        authorizer.setAllowListAllVolumes(allowListAllVolumes);
+      Class<? extends IAccessAuthorizer> clazz = configuration.getClass(
+          OZONE_ACL_AUTHORIZER_CLASS, OzoneAccessAuthorizer.class,
+          IAccessAuthorizer.class);
+      if (bNeedAccessAuthInitialization ||
+          clazz.getSimpleName().equals("OzoneNativeAuthorizer")) {
+        // In case of NativeAuthorizer always re-initialize
+        accessAuthorizer = getACLAuthorizerInstance(configuration, clazz);
+        if (accessAuthorizer instanceof OzoneNativeAuthorizer) {
+          OzoneNativeAuthorizer authorizer =
+              (OzoneNativeAuthorizer) accessAuthorizer;
+          isNativeAuthorizerEnabled = true;
+          authorizer.setVolumeManager(volumeManager);
+          authorizer.setBucketManager(bucketManager);
+          authorizer.setKeyManager(keyManager);
+          authorizer.setPrefixManager(prefixManager);
+          authorizer.setAdminCheck(ozoneManager::isAdmin);
+          authorizer.setReadOnlyAdminCheck(ozoneManager::isReadOnlyAdmin);
+          authorizer.setAllowListAllVolumes(allowListAllVolumes);
+        } else {
+          isNativeAuthorizerEnabled = false;
+        }
       } else {
+        accessAuthorizer = ozoneManager.getAccessAuthorizer();
```

Comment:
Looks like `accessAuthorizer` isn't closed or destroyed upon `OmMetadataReader` instance closure. Should be fine to share it like this IMO.